### ISSUE TITLE
Fragment with set:html before <head> element inserts to wrong place and removes the head element

### ISSUE THREAD

**Issue by susickypavel** [withastro/astro#6478]
### What version of `astro` are you using?

2.1.2

### Are you using an SSR adapter? If so, which one?

No

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### Describe the Bug

When Fragment with `set:html` or `set:text` is used before head element it breaks the HTML structure completely by removing head element in the final output and inserting the content of the fragment to the body.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-45dh9n-qyhnbj?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by susickypavel** [comment]
I wanted to insert a comment about the current version of the page + build time so that I can keep track of my releases. It's more of a visual aspect. I moved the comment inside the head element, and that works ok. The very top of the document would be just a tiny bit more convenient.

**Comment by matthewp** [comment]
Can you use a regular comment node `<!-- -->` without the fragment?

**Comment by susickypavel** [comment]
I'm not sure how. The content of the fingerprint is generated using dynamic data. 

```tsx
const BUILD_FINGERPRINT = `<!-- v${version} : ${process.env.ENVIRONMENT} - ${new Date().toUTCString()} -->`;
```

If i did this

```html
<!-- v${version} : ${process.env.ENVIRONMENT} - ${new Date().toUTCString()} -->
```

It's treated correctly as a comment.

**Comment by hjonasson** [comment]
I just verified that all content from `head` get moved to `body` for some reason. @susickypavel 's use case might not be the most important to fix since they found a solution but it does look like a strange bug. I spent a little time looking at it and ended up in [this file](https://github.com/withastro/astro/blob/8ce40a417c854d9e6a4fa7d5a85d50a6436b4a3c/packages/astro/src/runtime/server/render/common.ts#L12). If someone can confirm I am on the right path I can spend a bit of time on this next week.
